pygmt/datasets/load_earth_dataset.py

```diff
@@ -0,0 +1,208 @@
+"""
+Internal function to load GMT remote datasets.
+"""
+from typing import Dict, NamedTuple
+
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import kwargs_to_strings
+from pygmt.io import load_dataarray
+from pygmt.src import grdcut, which
+
+
+class Resolution(NamedTuple):
+    """
+    The available grid registrations for a given resolution and whether it is a
+    tiled grid.
+
+    Attributes
+    ----------
+    registrations : list
+        A list of the accepted registrations for a given resolution.
+        Can be either "pixel" or "gridline".
+
+    tiled : bool
+        States if the given resolution is tiled, which requires an
+        argument for ``region``."
+    """
+
+    registrations: list[str]
```

Comment:
There is an error here. 
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/runner/work/pygmt/pygmt/pygmt/__init__.py", line 25, in <module>
    from pygmt import datasets
  File "/home/runner/work/pygmt/pygmt/pygmt/datasets/__init__.py", line 5, in <module>
    from pygmt.datasets.earth_age import load_earth_age
  File "/home/runner/work/pygmt/pygmt/pygmt/datasets/earth_age.py", line 7, in <module>
    from pygmt.datasets.load_earth_dataset import _load_earth_dataset
  File "/home/runner/work/pygmt/pygmt/pygmt/datasets/load_earth_dataset.py", line 12, in <module>
    class Resolution(NamedTuple):
  File "/home/runner/work/pygmt/pygmt/pygmt/datasets/load_earth_dataset.py", line 28, in Resolution
    registrations: list[str]
TypeError: 'type' object is not subscriptable
```

Reply:
This error only seems to be on the Python 3.8 tests; is the solution to  just say that it is a list without any mention of the string?

Reply:
> This error only seems to be on the Python 3.8 tests.

No, it should also fail for the Python 3.10 tests. The Python 3.10 tests crash earlier due to https://github.com/GenericMappingTools/pygmt/issues/2215. 

> is the solution to just say that it is a list without any mention of the string?

I don't know. The syntax is new to me.

Reply:
The `list[str]` type-hint was only added in Python 3.9, so it won't work on Python 3.8. See https://peps.python.org/pep-0585/.

Reply:
Shortened it to `list`.